flink-runtime/src/main/scala/org/apache/flink/runtime/jobmanager/JobManager.scala

```diff
@@ -1358,6 +1358,10 @@ class JobManager(
                   throw new SuppressRestartsException(e)
               }
             }
+            else {
```

Comment:
If we did not have the unconditional initialization logic, we should also be able to drop this part. That would also make this work with the flip-6 code.